src/drivers/github.go

```diff
@@ -8,21 +8,46 @@ import (
 
 	"golang.org/x/oauth2"
 
-	"github.com/git-town/git-town/src/git"
 	"github.com/google/go-github/github"
 )
 
+type githubConfig interface {
+	GetCodeHostingDriverName() string
+	GetMainBranch() string
+	GetURLRepositoryName(string) string
+	GetURLHostname(string) string
+	GetCodeHostingOriginHostname() string
+	GetRemoteOriginURL() string
+	GetGitHubToken() string
+}
+
```

Comment:
At least
```golang
GetCodeHostingDriverName() string
	GetURLRepositoryName(string) string
	GetURLHostname(string) string
	GetCodeHostingOriginHostname() string
	GetRemoteOriginURL() string
```

are comon to all and could go into core.go

Actually those:
```
	GetURLRepositoryName(string) string
	GetURLHostname(string) string
```

are just utility functions, no methods at all and should be moved into the drivers package. So zero that out, as well, from the interface.

Reply:
In my current thinking, this `githubConfig` interface describes the methods that the GitHub driver will call in order to get the configuration values it needs. At least one of them (`GetGitHubToken`) is specific to this `githubCodeHostingDriver` class.

You are right that the utility functions don't really belong into the `git.Configuration` struct. That's a separate cleanup.